Misleading documentation note for NC_FORMATX_NCZARR

* the version of the software with which you are encountering an issue
Current main

* a description of the issue with the steps needed to reproduce it

The [documentation](https://github.com/Unidata/netcdf-c/blob/411f791dcaf5c8dc87f2cba20cb2aeea86917f41/include/netcdf.h#L199) for the NC_FORMATX_NCZARR states that it was "Added in version 4.3.1", however that is not accurate.
Instead, NC_FORMATX_ZARR was added in 4.3.1, and then the NC_FORMATX_ZARR symbol was renamed to NC_FORMAT_NCZARR released as part of 4.8.0 (renamed in [59e04a](https://github.com/Unidata/netcdf-c/commit/59e04ae0712a59688e5ae80dfbe7fc4c295d2826)).

This issue surfaced since version 1.7.1post2 of the python module uses this symbol, and version 4.7.0 of the C library was on the system, but we got:
`external/nc_complex/src/nc_complex.c:637:21: error: ‘NC_FORMATX_NCZARR’ undeclared (first use in this function)`

Looking up the symbol suggested it was introduced in 4.3.1, and given that 4.7.0 was on the system led to some confusion.

--- a/include/netcdf.h
+++ b/include/netcdf.h
@@ -231,7 +231,7 @@ Use this in mode flags for both nc_create() and nc_open(). */
 #define NC_FORMATX_DAP4      (6)
 #define NC_FORMATX_UDF0      (8)
 #define NC_FORMATX_UDF1      (9)
-#define NC_FORMATX_NCZARR    (10)
+#define NC_FORMATX_NCZARR    (10) /**< Added in version 4.8.0 */
 #define NC_FORMATX_UNDEFINED (0)
 
   /* To avoid breaking compatibility (such as in the python library),
@@ -247,6 +247,7 @@ Use this in mode flags for both nc_create() and nc_open(). */
 #define NC_FORMAT_DAP2      NC_FORMATX_DAP2 /**< \deprecated As of 4.4.0, use NC_FORMATX_DAP2 */
 #define NC_FORMAT_DAP4      NC_FORMATX_DAP4 /**< \deprecated As of 4.4.0, use NC_FORMATX_DAP4 */
 #define NC_FORMAT_UNDEFINED NC_FORMATX_UNDEFINED /**< \deprecated As of 4.4.0, use NC_FORMATX_UNDEFINED */
+#define NC_FORMATX_ZARR     NC_FORMATX_NCZARR /**< \deprecated as of 4.8.0, use NC_FORMATX_NCZARR */
 
 /**@}*/
 

